docker/CMSRucioClient/scripts/legacydn_converter.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+
+"""legacydn_converter.py
+
+A script that retrieves identities from a Rucio instance, converts it to
+an RFC2253 compliant DN, then adds it back to the account identity.
+"""
+
+import argparse
+from itertools import groupby
+import os
+
+from pprint import pprint
+
+from rucio.client.accountclient import AccountClient
+
+PROXY = os.getenv('X509_USER_PROXY')
+
+RFC_ATTRIBUTE_ORDER = ["CN", "L", "ST", "O", "OU", "C", "STREET", "DC", "UID"]
+
+client = AccountClient()
+
+
+def get_attribute(attribute: str) -> str:
+    """Extracts attribute key"""
+    try:
+        return attribute[:attribute.index("=")]
+    except ValueError:
+        # print(f"element lacks attribute, {attribute}, {dn}")
+        return ""
+
+
+def rfc2253dn(legacy_dn: str, verbose: bool = False) -> str:
+    """Converts legacy slash DB to comma separated DN"""
+    if not legacy_dn.startswith('/'):
+        if verbose:
+            print(f'DN does not start with /: {legacy_dn}')
+        return legacy_dn
+
+    # replace commas with an escape character
+    legacy_dn = legacy_dn.replace(',', r'\,')
+    parts = legacy_dn.split('/')[1:]
+
+    # parts are reversed. See https://datatracker.ietf.org/doc/html/rfc2253#section-2.3
+    elements = parts[::-1]
```

Comment:
IIUC, this should already be correct here, right? Is the expected order part needed? Do we have DNs which are malformed in the legacy format which this is correcting?

Reply:
I did find a few DNs that had the order of attributes slightly off compared to the table in the RFC document (https://datatracker.ietf.org/doc/html/rfc2253#section-2.3). The RFC doesn't explicitly say that it has to be in the order in the table, but I see that the table isn't sorted alphabetically. My personal cert didn't fall under this case, so I can't really test this.

Assuming Apache is compliant, the DN sent to Rucio for verification should be in the order defined in the RFC.

Reply:
OK great. Let me know when you've got this ready to merge.

Reply:
I will save a list of the DNs that don't match the order in the table so we can monitor it once we decide to stop using legacy DNs.